src/node/server/serverPluginCss.ts

```diff
@@ -175,14 +177,15 @@ export const cssPlugin: ServerPlugin = ({ root, app, watcher, resolver }) => {
 export function codegenCss(
   id: string,
   css: string,
-  modules?: Record<string, string>
+  modules?: Record<string, string>,
+  modulesNamedExports = false
 ): string {
   let code =
     `import { updateStyle } from "${clientPublicPath}"\n` +
     `const css = ${JSON.stringify(css)}\n` +
     `updateStyle(${JSON.stringify(id)}, css)\n`
   if (modules) {
-    code += `export default ${JSON.stringify(modules)}`
+    code += codegenCssModules(modules, modulesNamedExports)
```

Comment:
You can use same logic with https://github.com/vitejs/vite/commit/353d5a2f29b7981948c5bb3a752fc1e8742c6598 to transform json to es module.

Reply:
Ooh, didn't know that, will resolve in a bit.

Reply:
It seems that `dataToEsm` skips over keys that aren't valid JS identifiers. Should camel casing be enforced, or should it throw an error if `localsConvention` isn't camel case?

Should the namedExports option even exists, when the JSON transformer enforces named exports? :joy:

Reply:
It's unnecssary for add `namedExports ` option.  Transformed it if `localsConvention ` is `camelCase`, else don‘t transform it (only export  default).

Reply:
Alright.